lib/python/ray/worker.py

```diff
@@ -526,6 +533,7 @@ def remote(arg_types, return_types, worker=global_worker):
     return_types (List[type]): List of Python types of the return values.
   """
   def remote_decorator(func):
+    to_export = cloudpickle.dumps((func, arg_types, return_types)) if worker.mode == ray.SHELL_MODE else None
```

Comment:
probably `if worker.mode == ray.SHELL_MODE or worker.mode == ray.SCRIPT_MODE`, basically whenever the worker is a driver